test/test_jit.py

```diff
@@ -14168,6 +14168,26 @@ def test():
         with self.assertRaisesRegex(Exception, "Parameters not specified"):
             torch.jit.script(test)
 
+        def wrong_decl_body(x: str) -> str:
+            return x + "0"
+
+        with self.assertRaisesRegex(RuntimeError, "Only `pass` statement can be the body"):
+            torch.jit._overload(wrong_decl_body)
+
+        @torch.jit._overload
+        def null_overload(x: int) -> int:
+            pass
+
+        @torch.jit._overload
+        def null_overload(x: str) -> str:  # noqa: F811
+            pass
+
+        def null_overload_driver():
+            return null_overload(0)
+
+        with self.assertRaisesRegex(RuntimeError, 'Implementation for the function ".+" is missing.'):
```

Comment:
What about the case, say there is a later definition of `null_overload` but its signature mismatches that of its overload declarations?

Reply:
we can add another check to test whether decl and def have the same # of arguments, I guess that's what you mean?